src/System.ServiceModel.Primitives/tests/Channels/UnderstoodHeadersTest.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Infrastructure.Common;
+using System;
+using System.Collections;
+using System.Linq;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using Xunit;
+
+public static class UnderstoodHeadersTest
+{
+    [WcfFact]
+    public static void UnderstoodHeaders_Add_Contains_Remove_GetEnumerator()
+    {
+        // *** SETUP *** \\
+        BasicHttpBinding binding = new BasicHttpBinding();
+
+        // Create Message object
+        Message requestMessage = Message.CreateMessage(
+            binding.MessageVersion,
+            "http://tempuri.org/AnyService",
+            new CustomBodyWriter("Any Message"));
+
+        // Create additional message headers.
+        MessageHeader customHeaderAlpha = MessageHeader.CreateHeader("AlphaHeader", "AlphaHeaderNamespace", "AlphaObjectValue");
```

Comment:
Purely a nit, feel free to ignore. Strictly speaking, a namespace should be of the form http://... so maybe make it `http://tempuri.org/AlphaHeaderNamespace`.